cjwstate/modules/i18n/catalogs.py

```diff
@@ -0,0 +1,75 @@
+from cjworkbench.i18n import supported_locales, default_locale
+from cjworkbench.i18n.catalogs.util import (
+    read_po_catalog,
+    write_po_catalog,
+    remove_strings,
+    find_fuzzy_messages,
+    fill_catalog,
+    mark_fuzzy,
+    MessageUID,
+    catalogs_are_same,
+)
+from babel.messages.catalog import Catalog
+from cjwstate.modules.module_loader import ModuleFiles, ModuleSpec
+from typing import FrozenSet, Dict
+import pathlib
+from cjwstate.modules.i18n.spec import find_spec_messages
+
+
+def update_module_catalogs(directory: pathlib.Path):
+    module_files = ModuleFiles.load_from_dirpath(directory)  # raise ValueError
+    source_catalog = _build_source_catalog(ModuleSpec.load_from_path(module_files.spec))
+
+    po_path = _po_path(directory, default_locale)
+
+    try:
+        old_source_catalog = read_po_catalog(po_path)
+    except FileNotFoundError:
+        old_source_catalog = Catalog(default_locale)
+
+    if not catalogs_are_same(source_catalog, old_source_catalog):
```

Comment:
I'm geeking out at how legible this code is :).

What happens in this scenario:

1. Module author publishes a module in English and French
2. Workbench enables Portuguese
3. Module author runs `update_module_catalogs()`

Expected results: this function writes a new `<module>/locale/pt/messages.po`
Actual results: it doesn't, because `<module>/messages.pot` is unchanged